monix-eval/shared/src/main/scala/monix/eval/TaskLift.scala

```diff
@@ -61,9 +63,22 @@ object TaskLift extends TaskLiftImplicits0 {
     */
   implicit def toIO(implicit eff: ConcurrentEffect[Task]): TaskLift[IO] =
     new TaskLift[IO] {
-      def taskLift[A](task: Task[A]): IO[A] =
-        task.toIO(eff)
+      def apply[A](task: Task[A]): IO[A] =
+        TaskConversions.toIO(task)(eff)
     }
+
+  /**
+    * Deprecated method, which happened on extending `FunctionK`.
+    */
+  implicit class Deprecated[F[_]](val inst: TaskLift[F]) {
+    /** DEPRECATED — switch to [[TaskLift.apply]]. */
+    @deprecated("Switch to TaskLift.apply", since = "3.0.0-RC3")
+    def taskLift[A](task: Task[A]): F[A] = {
```

Comment:
Why not have this as a final method on TaskLift that just delegates to `apply`? Same for all the `Deprecated` classes

Reply:
Hmm, maybe, thinking about it.